include/torch-mlir/Conversion/Passes.td

```diff
@@ -105,6 +105,59 @@ def ConvertTorchToLinalg : Pass<"convert-torch-to-linalg", "func::FuncOp"> {
   let constructor = "mlir::torch::createConvertTorchToLinalgPass()";
 }
 
+def ConvertTorchBackendLegalToTosaCustom
+    : Pass<"convert-torch-backend-legal-to-tosa-custom", "func::FuncOp"> {
+  let summary = "Convert Torch backend-legal operations to TOSA CustomOps.";
+  let description = [{
+    This pass extends the selective decomposition patterns to TOSA backend,
+    such that it will convert operations marked to custom-ops by the user
+    for the TOSA backend to CustomOps in TOSA dialect.
+    The backend-custom operations will not be decomposed or lowered through
+    the existing Torch to TOSA conversion patterns, even if such patterns
+    exist in the subsequent Torch to TOSA conversion pass; instead a TOSA
+    CustomOp will be created with the specifications below:
+      1. The 'identifier' attribute in the CustomOp will be the operation
+         name of the corresponding ATen operation.
+      2. The 'config' attribute in the CustomOp will be the specification
+         of the compilation pipeline, i.e. "torch_mlir"
+      3. All inputs to the ATen operation will be converted into legal TOSA
+         operations. There will be no distiction between inputs and constant
+         attributes; all will be treated as inputs to the CustomOp. It is up
+         to the consuming backend to deduce them based on the ATen operation
+         semantics.
+      4. Since TOSA conversion pattern of Torch operations are responsible
+         to consume and convet the constant attributes (such as 'axis' for a
+         reduction operation), the TOSA CustomOp conversion will also match
+         and rewrite these attributes as TOSA ConstOps as well. Specifically:
+           - torch.constant.bool        -> tosa.ConstOp (of tensor type i64)
+           - torch.constant.int         -> tosa.ConstOp (of tensor type i64)
+           - torch.constant.float       -> tosa.ConstOp (of tensor type f32)
+           - torch.prim.ListConstruct   -> tosa.ConstOp (of tensor type i64)
```

Comment:
I think it would be clearer to use the types rather than the ops here. This will allow being explicit about which lists are supported and which are not. I'm assuming support for lists here is just for `!torch.list<int>`, right?

Reply:
Done, updated to use types instead of operations. It is explicit that only list of integers are supported (i.e., `!torch.list<int>`)